### ISSUE TITLE
Exception when using GTA with VsTest.Console.exe (VS 2017 Version 15.5)

### ISSUE THREAD

**Issue by drocx** [csoltenborn/GoogleTestAdapter#184]
Hello,

the GoogleTestAdapter is crashing with Visual Studio 2017 Version 15.5 with the following error:

```
Microsoft (R) Testausführungs-Befehlszeilentool Version 15.5.0
Copyright (c) Microsoft Corporation. Alle Rechte vorbehalten.

Die Testausführung wird gestartet, bitte warten...
Ausnahme beim Aufrufen von Executor "executor://googletestrunner/v1": Der Objektverweis wurde nicht auf eine Objektinstanz festgelegt.
```

Steps to reproduce:

1. Install VS 2017 with the lastest Update via Visual Studio Installer
The file version number of the vstest.console.exe should be 15.0.27128.1.

2. Run the SampleTest Project 
 :information_source: I had to fix the following [warning ](https://docs.microsoft.com/de-de/cpp/cpp-conformance-improvements-2017)before i could compile the project:
```
warning C4996: 'std::tr1': warning STL4002: The non-Standard std::tr1 namespace and TR1-only machinery are deprecated and will be REMOVED. You can define _SILENCE_TR1_NAMESPACE_DEPRECATION_WARNING to acknowledge that you have received this warning.
```
3. Run the compiled SampleTest via Commandline:
```
"C:\Program Files (x86)\Microsoft Visual Studio\2017\Enterprise\Common7\IDE\CommonExtensions\Microsoft\TestWindow\vstest.console.exe" C:\Users\USERNAME\Downloads\GoogleTestAdapter-master\GoogleTestAdapter-master\out\binaries\SampleTests\Debug\CrashingTests_gta.exe /Settings:C:\Users\USERNAME\Downloads\GoogleTestAdapter-master\GoogleTestAdapter-master\SampleTests\SampleTests.gta.runsettings "/TestAdapterPath:\"C:\Users\USERNAME\Downloads\GoogleTestAdapter-master\GoogleTestAdapter-master""
```

If i remove the SampleTests.gta.runsettings line from the commandline call the test are working.

Regards
Daniel

**Comment by csoltenborn** [comment]
Thanks for the (great) bug report! I will look into this in a couple of days...

**Comment by csoltenborn** [comment]
Just had a quick glance at this problem: turns out that the command `vstest.console /TestAdapterPath:"...\GoogleTestAdapter\NuGetPackages" /ListExecutors /Settings:"...\GoogleTestAdapter\SampleTests\SampleTests.gta.runsettings"` works just fine against VS 15.4.5, but stopped working with VS 15.5. Thus, this looks like a VS regression to me...

@spebl @nickuhlenhuth Is this a known problem? Does TAfGT suffer from this? Any chance that one of you can have a look?

**Comment by csoltenborn** [comment]
Some more insights:
* The root problem appears to be that the `SettingsProvider` is not found any more, causing [this line](https://github.com/csoltenborn/GoogleTestAdapter/blob/7976c6e9d8e0fb362d859a98dfeb844dd0caafd9/GoogleTestAdapter/TestAdapter/CommonFunctions.cs#L39) to throw an exception (exception message: "Problem loading settings: Der Einstellungsanbieter namens "GoogleTestAdapterSettings" wurde nicht gefunden. Die Einstellungen können nicht geladen werden.")
  * This is consistent with the fact that the `Console_ListDiscoverers_DiscovererIsListed`, `Console_ListExecutors_ExecutorIsListed`, and `Console_ListSettingsProviders_SettingsProviderIsListed` tests fail when compiling GTA with VS 15.5 and then running the end-to-end tests.
* The `NullReferenceException` reported by VS is caused by [this line](https://github.com/csoltenborn/GoogleTestAdapter/blob/7976c6e9d8e0fb362d859a98dfeb844dd0caafd9/GoogleTestAdapter/TestAdapter/TestExecutor.cs#L53): The exception is caught properly but prevented the logger from being initialized; this causes the NRE.

**Comment by smadala** [comment]
@csoltenborn This issue looks like with making [Testplatform V2](https://github.com/Microsoft/vstest) default in VS15.5. 
Can you open issue on https://github.com/Microsoft/vstest with link to detailed sample repro?

[Related issue](https://developercommunity.visualstudio.com/content/problem/224700/unable-to-run-specflow-tests-using-runsettings-fil.html)?

**Comment by csoltenborn** [comment]
@smadala I have opened an issue (see link above). Thanks for looking into the issue, and let me know if you need any further information (here or at the new issue)...

**Comment by csoltenborn** [comment]
Here's some good news: Looks like [the actual issue](https://github.com/Microsoft/vstest/issues/1652) is going to be fixed soon.

**Comment by smadala** [comment]
Please use `/Inisolation` option to mitigate the issue.

**Comment by csoltenborn** [comment]
Thanks for the workaround! However, for GTA it's also possible (since 0.13.0) to use n environment variable to specify the location of the settings file - GTA will then just load them from that place.

**Comment by csoltenborn** [comment]
This issue has apparently been fixed for V15.9.0 of VsTest. However, note that VS2017 at V15.8.7 still comes with VsTest V15.8.0. If you urgendtly need this fix, consume VsTest.Console via [NuGet](https://www.nuget.org/packages/Microsoft.TestPlatform).

**Comment by csoltenborn** [comment]
This issue has apparently been fixed for V1.5.9 of VTest. However, VS 15.8.7 still comes with V15.8.0; thus, if you urgently need this fix, consume VsTest.Console via [NuGet]()https://www.nuget.org/packages/Microsoft.TestPlatform
